libvirt/resource_libvirt_domain.go

```diff
@@ -149,60 +151,74 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 		netIface.Mac.Address = mac
 
-		// this is not passes to libvirt, but used by waitForAddress
+		// this is not passed to libvirt, but used by waitForAddress
+		netIface.waitForLease = false
 		if waitForLease, ok := d.GetOk(prefix + ".wait_for_lease"); ok {
 			netIface.waitForLease = waitForLease.(bool)
 		}
 
-		networkName := "default"
+		// connect to the interface to the network... first, look for the network
 		if n, ok := d.GetOk(prefix + ".network_name"); ok {
 			// when using a "network_name" we do not try to do anything: we just
 			// connect to that network
-			networkName = n.(string)
+			netIface.Type = "network"
+			netIface.Source.Network = n.(string)
 		} else if networkUUID, ok := d.GetOk(prefix + ".network_id"); ok {
+			// when using a "network_id" we are referring to a "network resource"
+			// we have defined somewhere else...
 			network, err := virConn.LookupNetworkByUUIDString(networkUUID.(string))
 			if err != nil {
 				return fmt.Errorf("Can't retrieve network ID %s", networkUUID)
 			}
-			networkName, err = network.GetName()
+			networkName, err := network.GetName()
 			if err != nil {
 				return fmt.Errorf("Error retrieving volume name: %s", err)
 			}
-			getOrEmpty := func(field string) string {
-				if r, ok := d.GetOk(prefix + "." + field); ok {
-					return r.(string)
-				} else {
-					return ""
-				}
-			}
-
-			// try to associate the IP / hostname to the MAC in that network
-			address, hostname := getOrEmpty("address"), getOrEmpty("hostname")
-			if len(address) > 0 {
-				// TODO: check the address is in the valid ranges for this network
 
-				if len(hostname) == 0 {
-					// try to use the name from the domain definition when no hostname has been specified
-					hostname = domainDef.Name
+			hostname := domainDef.Name
+			if hostnameI, ok := d.GetOk(prefix + ".hostname"); ok {
+				hostname = hostnameI.(string)
+			}
+			if addresses, ok := d.GetOk("addresses"); ok {
+				// some IP(s) provided
```

Comment:
I think we should fail if an IP address is provided for an interface attached to a bridged network. There's no control over the IP address in that case